spec/shared_contexts.rb

```diff
@@ -48,3 +48,109 @@ def get_plugin_content(command_class_name, command_name)
     plugin_renderer.render
   end
 end
+
+RSpec.shared_context :allow_net_connect do
+  around :context do |example|
+    WebMock.allow_net_connect!
+    example.run
+    WebMock.disable_net_connect!
+  end
+end
+
+RSpec.shared_context :real_api3scale_client do
+  include_context :allow_net_connect
+  include_context :random_name
+
+  puts '================ RUNNING REAL 3SCALE API CLIENT ========='
+
+  let(:endpoint) { ENV.fetch('ENDPOINT') }
+
+  let(:provider_key) { ENV.fetch('PROVIDER_KEY') }
+
+  let(:verify_ssl) { !(ENV.fetch('VERIFY_SSL', 'true').to_s =~ /(true|t|yes|y|1)$/i).nil? }
+
+  let(:http_client) do
+    ThreeScale::API::HttpClient.new(endpoint: endpoint,
+                                    provider_key: provider_key,
+                                    verify_ssl: verify_ssl)
+  end
+  let(:api3scale_client) { ThreeScale::API::Client.new(http_client) }
+end
+
+class OpenAPITemplateRenderer
+  attr_accessor :title
+
+  def initialize(template)
+    @renderer = ERB.new(template)
+  end
+
+  def render
+    @renderer.result(binding)
+  end
+end
+
+RSpec.shared_context :import_oas_real_cleanup do
+  after :each do
+    service.delete_service
+  end
+end
+
+RSpec.shared_context :import_oas_stubbed_api3scale_client do
+  puts '============ RUNNING STUBBED 3SCALE API CLIENT =========='
```

Comment:
This is not printed when running the client but when importing the module. It would have to be within `before(:all)` block.